### PR TITLE
ci: create complex action with multiple nodes

### PR DESCRIPTION
**Pull request by BlaineEXE**
Create a complex CI action with multiple minikube nodes. Add multiple
virtual disks to each and additional NICs to each. This will enable
complex CI tests for cases like Multus or arbiter.

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: .github/workflows/complex.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    # manusa/actions-setup-minikube only works in ubuntu, but macos is the only github runner that
+    # supports nested virtualization
+    # macos VM specs: 3 cores of CPU, 14G RAM, 14G SSD space
+    - name: setup minikube cluster
+      run: |
+        NUM_DISKS_PER_VM=3
+        EXTRA_DISK_SIZE_MB=10000
+
+        time minikube start --driver=virtualbox --memory=6g --nodes=3
```
Comment (leseb) [inline .github/workflows/complex.yaml:41]: See:

```
Hardware specification for macOS virtual machines:

3-core CPU
14 GB of RAM memory
14 GB of SSD disk space
```

So we can bump the memory a bit I think.

Comment (BlaineEXE) [reply .github/workflows/complex.yaml:41]: 6G from 3 nodes is 18G. I don't think we want to over-provision the host by too much, do we?

Comment (leseb) [reply .github/workflows/complex.yaml:41]: Oh right, 3 nodes. Sounds good to keep it like this.

**Comment by sp98** [comment]
https://github.com/marketplace/actions/absaoss-k3d-action  could be an option worth exploring.  Lightweight and provides options like multiple clusters and private registry.

**Comment by BlaineEXE** [comment]
> https://github.com/marketplace/actions/absaoss-k3d-action could be an option worth exploring. Lightweight and provides options like multiple clusters and private registry.

This looks great. I wonder how we might get this to work with virtual hard disks though... perhaps loop devices?

One of the goals of this exploration was to also see about adding extra virtualized network interfaces to test multus, which I don't think we'll be able to do easily without full virtualization. I'm happy to look at alternative suggestions on this topic, however.

**Pull request closed without merging**